platform/darwin/src/MGLGeoJSONSource.mm

```diff
@@ -28,6 +30,7 @@ - (instancetype)initWithIdentifier:(NSString *)identifier geoJSONData:(NSData *)
     {
         _geoJSONData = data;
         _options = options;
+        [self commonInit];
```

Comment:
`-[MGLSource initWithIdentifier:]` should implement a private `-commonInit` stub so that `-[MGLSource initWithIdentifier:]` should call it and consolidate all these disparate calls. Each MGLSource subclass can still override the method to do things specific to the particular kind of `mbgl::style::Source` it’s working with.